op-challenger/game/fault/solver/rules.go

```diff
@@ -67,13 +76,113 @@ func doNotDuplicateExistingMoves(game types.Game, action types.Action) error {
 	return nil
 }
 
-func doNotDefendRootClaim(game types.Game, action types.Action) error {
+func doNotDefendRootClaim(game types.Game, action types.Action, correctTrace types.TraceProvider) error {
 	if game.Claims()[action.ParentIdx].IsRootPosition() && !action.IsAttack {
 		return fmt.Errorf("defending the root claim at idx %v", action.ParentIdx)
 	}
 	return nil
 }
 
+func avoidPoisonedPrestate(game types.Game, action types.Action, correctTrace types.TraceProvider) error {
+	if action.Type == types.ActionTypeStep {
+		return nil
+	}
+	ancestors := ""
+	movePosition := resultingPosition(game, action)
+	honestTraceIndex := movePosition.TraceIndex(game.MaxDepth())
+	// Walk back up the claims and find the claim with highest trace index < honestTraceIndex
+	claim := game.Claims()[action.ParentIdx]
+	var preStateClaim types.Claim
+	for {
+		ancestors += printClaim(claim, game) + "\n"
+		claimTraceIdx := claim.TraceIndex(game.MaxDepth())
+		if claimTraceIdx.Cmp(honestTraceIndex) < 0 { // Check it's left of the honest claim
+			if preStateClaim == (types.Claim{}) || claimTraceIdx.Cmp(preStateClaim.TraceIndex(game.MaxDepth())) > 0 {
+				preStateClaim = claim
+			}
+		}
+		if claim.IsRoot() {
+			break
+		}
+		parent, err := game.GetParent(claim)
+		if err != nil {
+			return fmt.Errorf("no parent of claim %v: %w", claim.ContractIndex, err)
+		}
+		claim = parent
+	}
+	if preStateClaim == (types.Claim{}) {
+		// No claim to the left of the honest claim, so can't have been poisoned
+		return nil
+	}
+	correctValue, err := correctTrace.Get(context.Background(), preStateClaim.Position)
+	if err != nil {
+		return fmt.Errorf("failed to get correct trace at position %v: %w", preStateClaim.Position, err)
+	}
+	if correctValue != preStateClaim.Value {
+		// How far left can we push this claim?
+		pos := movePosition
+		for pos.Depth() < game.MaxDepth() {
+			pos = pos.Attack()
+		}
+		err = fmt.Errorf("prestate poisoned claim %v has invalid prestate and is left of honest claim countering %v at trace index %v", preStateClaim.ContractIndex, action.ParentIdx, honestTraceIndex)
+		fmt.Printf("%v\n%v\nAttack would get to %v\n", err.Error(), ancestors, pos.TraceIndex(game.MaxDepth()))
+		//return err
```

Comment:
Note: Not currently returning this error to let the test continue through and see if the `detectPoisonedStep` rule catches things at the end of the game where I'm more confident in the detection logic.  I _think_ this is right though and is actually detecting the first move that leaves the challenger vulnerable.